lora-phy/src/lib.rs

```diff
@@ -268,12 +300,23 @@ where
                         return Err(err);
                     }
                 }
+                self.wait_for_irq().await?;
```

Comment:
What are the effects of moving this to the end of the loop? I guess we would match Ok(None) but I am not sure

Reply:
Matching Ok(None) is what happens, yeah.

I wish I'd been smart enough to comment that change, as I wrote this months ago and can't remember the motivation. I think it had something to do with the irq already being cleared if you'd called wait_for_packet_start, but I'll have to dig in and do some testing to be sure.

Reply:
Generally I think it is probably better after, as you wrote it, to avoid edge cases. "Clearing the queue" so to say but this is just a gut feeling.